# Added preference for disabling Restoring to previous zoom value on page
# Internal requirements

Index: mozilla/docshell/base/nsDocShell.cpp
===================================================================
--- mozilla.orig/docshell/base/nsDocShell.cpp
+++ mozilla/docshell/base/nsDocShell.cpp
@@ -5406,17 +5406,22 @@
     }
 
     nsCOMPtr<nsIMarkupDocumentViewer> oldMUDV(do_QueryInterface(mContentViewer));
     nsCOMPtr<nsIMarkupDocumentViewer> newMUDV(do_QueryInterface(viewer));
     float textZoom = 1.0f;
     float pageZoom = 1.0f;
     if (oldMUDV && newMUDV) {
         oldMUDV->GetTextZoom(&textZoom);
-        oldMUDV->GetFullZoom(&pageZoom);
+        PRBool preserveZoom = PR_FALSE;
+        mPrefs->GetBoolPref("browser.shistory.preservezoom", &preserveZoom);
+        if (preserveZoom)
+            newMUDV->GetFullZoom(&pageZoom);
+        else
+            oldMUDV->GetFullZoom(&pageZoom);
     }
 
     // Protect against mLSHE going away via a load triggered from
     // pagehide or unload.
     nsCOMPtr<nsISHEntry> origLSHE = mLSHE;
 
     // Notify the old content viewer that it's being hidden.
     FirePageHideNotification(!mSavingOldViewer);
